ts/main.ts

```diff
@@ -0,0 +1,44 @@
+interface FormElements extends HTMLFormControlsCollection {
+  title: HTMLInputElement;
+  photoURL: HTMLInputElement;
+  notes: HTMLInputElement;
+}
+interface FormEntry {
+  entryID: number;
+  title: string;
+  photoURL: string;
+  notes: string;
+}
+
+const $photoInput = document.querySelector('.photo-input') as HTMLInputElement;
+if (!$photoInput) throw new Error('$photoInput not found');
+
+const $photoPreview = document.querySelector(
+  '.photo-preview',
+) as HTMLImageElement;
+if (!$photoPreview) throw new Error('$photoPreview not found');
+
+$photoInput.addEventListener('input', (event: Event) => {
+  const $input = event.target as HTMLInputElement;
+  $photoPreview.src = $input.value;
+  writeData();
```

Comment:
Not needed. You didn't do anything to your `data` variable. We only call `writeData` when you manipulate `data`